packages/hub/routes/merchant-infos.ts

```diff
@@ -64,6 +72,50 @@ export default class MerchantInfosRoute {
     ctx.body = serialized;
     ctx.type = 'application/vnd.api+json';
   }
+
+  async getValidation(ctx: Koa.Context) {
+    let slug = ctx.query?.slug;
+    let isValidSlug = await ensureValidSlug(ctx, slug, this.merchantInfoQueries);
+
+    if (!isValidSlug) {
+      return;
+    }
+
+    ctx.status = 200;
+    ctx.body = {
+      slugAvailable: true,
+      info: 'Merchant slug is available',
+    };
+    ctx.type = 'application/vnd.api+json';
+  }
+}
+
+async function ensureValidSlug(ctx: Koa.Context, slug?: string | string[], queries?: MerchantInfoQueries) {
```

Comment:
This function is outside of the class, so that `this` doesn't apply here, but I can very easily move it inside the class. The queries being optional was a typescript quirk (because the argument that comes before it is optional, it gave an error). I can move the slug to be the last argument, which would satisfy typescript and not make queries optional.

Reply:
Oh right that makes sense.